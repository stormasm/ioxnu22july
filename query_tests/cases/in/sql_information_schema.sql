-- IOX_SETUP: TwoMeasurementsManyFields

-- validate we have access to information schema for listing table names (w/o system tables)
-- IOX_COMPARE: sorted
SELECT * from information_schema.tables where table_schema != 'system';

-- validate we have access to information schema for listing columns names/types
-- IOX_COMPARE: sorted
SELECT * from information_schema.columns where table_name = 'h2o' OR table_name = 'o2';

-- validate we have access to SHOW SCHEMA for listing columns names
-- IOX_COMPARE: sorted
SHOW COLUMNS FROM h2o;
