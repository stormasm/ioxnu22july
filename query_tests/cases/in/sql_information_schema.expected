-- Test Setup: TwoMeasurementsManyFields
-- SQL: SELECT * from information_schema.tables where table_schema != 'system';
-- Results After Sorting
+---------------+--------------------+------------+------------+
| table_catalog | table_schema       | table_name | table_type |
+---------------+--------------------+------------+------------+
| public        | information_schema | columns    | VIEW       |
| public        | information_schema | tables     | VIEW       |
| public        | information_schema | views      | VIEW       |
| public        | iox                | h2o        | BASE TABLE |
| public        | iox                | o2         | BASE TABLE |
+---------------+--------------------+------------+------------+
-- SQL: SELECT * from information_schema.columns where table_name = 'h2o' OR table_name = 'o2';
-- Results After Sorting
+---------------+--------------+------------+-------------+------------------+----------------+-------------+-----------------------------+--------------------------+------------------------+-------------------+-------------------------+---------------+--------------------+---------------+
| table_catalog | table_schema | table_name | column_name | ordinal_position | column_default | is_nullable | data_type                   | character_maximum_length | character_octet_length | numeric_precision | numeric_precision_radix | numeric_scale | datetime_precision | interval_type |
+---------------+--------------+------------+-------------+------------------+----------------+-------------+-----------------------------+--------------------------+------------------------+-------------------+-------------------------+---------------+--------------------+---------------+
| public        | iox          | h2o        | city        | 0                |                | YES         | Dictionary(Int32, Utf8)     |                          |                        |                   |                         |               |                    |               |
| public        | iox          | h2o        | moisture    | 1                |                | YES         | Float64                     |                          |                        | 24                | 2                       |               |                    |               |
| public        | iox          | h2o        | other_temp  | 2                |                | YES         | Float64                     |                          |                        | 24                | 2                       |               |                    |               |
| public        | iox          | h2o        | state       | 3                |                | YES         | Dictionary(Int32, Utf8)     |                          |                        |                   |                         |               |                    |               |
| public        | iox          | h2o        | temp        | 4                |                | YES         | Float64                     |                          |                        | 24                | 2                       |               |                    |               |
| public        | iox          | h2o        | time        | 5                |                | NO          | Timestamp(Nanosecond, None) |                          |                        |                   |                         |               |                    |               |
| public        | iox          | o2         | city        | 0                |                | YES         | Dictionary(Int32, Utf8)     |                          |                        |                   |                         |               |                    |               |
| public        | iox          | o2         | reading     | 1                |                | YES         | Float64                     |                          |                        | 24                | 2                       |               |                    |               |
| public        | iox          | o2         | state       | 2                |                | YES         | Dictionary(Int32, Utf8)     |                          |                        |                   |                         |               |                    |               |
| public        | iox          | o2         | temp        | 3                |                | YES         | Float64                     |                          |                        | 24                | 2                       |               |                    |               |
| public        | iox          | o2         | time        | 4                |                | NO          | Timestamp(Nanosecond, None) |                          |                        |                   |                         |               |                    |               |
+---------------+--------------+------------+-------------+------------------+----------------+-------------+-----------------------------+--------------------------+------------------------+-------------------+-------------------------+---------------+--------------------+---------------+
-- SQL: SHOW COLUMNS FROM h2o;
-- Results After Sorting
+---------------+--------------+------------+-------------+-----------------------------+-------------+
| table_catalog | table_schema | table_name | column_name | data_type                   | is_nullable |
+---------------+--------------+------------+-------------+-----------------------------+-------------+
| public        | iox          | h2o        | city        | Dictionary(Int32, Utf8)     | YES         |
| public        | iox          | h2o        | moisture    | Float64                     | YES         |
| public        | iox          | h2o        | other_temp  | Float64                     | YES         |
| public        | iox          | h2o        | state       | Dictionary(Int32, Utf8)     | YES         |
| public        | iox          | h2o        | temp        | Float64                     | YES         |
| public        | iox          | h2o        | time        | Timestamp(Nanosecond, None) | NO          |
+---------------+--------------+------------+-------------+-----------------------------+-------------+
